File: shared/src/types/resource.ts
export type Resource = {
  ID: number;
  id?: number;
  course_id: number;
  chapter_id?: number | null;
  created_by_id?: number;
  title?: string;
  name?: string;
  type?: 'video' | 'paper' | 'link' | string;
  url?: string;
  description?: string;
  file_url?: string;
  file_type?: string;
  file_size?: number;
  CreatedAt?: string;
  UpdatedAt?: string;
};

export type CreateResourceRequest = {
  course_id: number;
  title: string;
  type: 'video' | 'paper' | 'link';
  url: string;
  description?: string;
};

File: backend/internal/models/resource.go
package models

import "gorm.io/gorm"

// Resource represents a course resource (video, paper, link)
type Resource struct {
	gorm.Model
	CourseID    uint   `gorm:"not null;index" json:"course_id"`
	ChapterID   *uint  `gorm:"index" json:"chapter_id,omitempty"` // nullable, relates to chapter
	CreatedByID uint   `gorm:"not null;index" json:"created_by_id"`
	Title       string `gorm:"size:256;not null" json:"title"`
	Type        string `gorm:"size:32;not null" json:"type"` // video, paper, link
	URL         string `gorm:"size:1024;not null" json:"url"`
	Description string `gorm:"type:text" json:"description,omitempty"`
}
